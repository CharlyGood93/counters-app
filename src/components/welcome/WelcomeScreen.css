.color-text {
    color: var(--dark-black);
}