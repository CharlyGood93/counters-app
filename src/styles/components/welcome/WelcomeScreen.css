.container-fluid {
    text-align: center;
}

.col svg {
    margin: 100px 0px 80px 0px;
}

.container-fluid__title {
    margin-bottom: 50px;
}

.col .cs-loading-spinner {
    left: 50%;
    position: fixed;
    top: 50%;
    transform: translate(-50%, -50%);
}