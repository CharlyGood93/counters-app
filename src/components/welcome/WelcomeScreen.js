import React from 'react';
import { Col, Container, Row } from 'react-bootstrap';
import { useHistory } from 'react-router-dom';

import { ReactComponent as Logo } from '../../logo.svg';
import { Button } from '../../ui';

import '../../styles/components/welcome/WelcomeScreen.css';

export const WelcomeScreen = () => {

    const history = useHistory();

    const handleContinue = () => {
        history.push('/counters');
    }

    return (
        <Container fluid>
            <Row>
                <Col>
                    <Logo />
                </Col>
            </Row>
            <Row>
                <Col className="container-fluid__title">
                    <h4>
                        <strong>Welcome to Counters</strong>
                    </h4>
                    <p>Capture cups of lattes, frappucinos, or anything else that can be counted.</p>
                </Col>
            </Row>
            <Row>
                <Col>
                    <Button
                        onClick={handleContinue}>
                        Get started
                    </Button>
                </Col>
            </Row>
        </Container>
    )
}
