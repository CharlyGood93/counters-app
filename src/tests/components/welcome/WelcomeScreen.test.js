import React from 'react';
import { shallow } from "enzyme";
import { WelcomeScreen } from "../../../components/welcome/WelcomeScreen";

const wrapper = shallow(<WelcomeScreen />);
const textToCompare = 'Welcome to Counters';

describe('Test in <WelcomeScreen />', () => {
    test('Should show the <WelcomeScreen /> component correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
    test('Should have the title Welcome to Counters', () => {
       const title = wrapper.find('h4.color-text');
       expect(title.props('strong').children.props.children).toBe(textToCompare); 
    });
});
