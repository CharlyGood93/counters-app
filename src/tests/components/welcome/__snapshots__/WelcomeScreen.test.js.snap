// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <WelcomeScreen /> Should show the <WelcomeScreen /> component correctly 1`] = `
<Container
  fluid={false}
>
  <Row
    className="mt-5 py-5"
    noGutters={false}
  >
    <Col
      className="col-md-12 text-center"
    >
      <ForwardRef(SvgLogo) />
    </Col>
  </Row>
  <Row
    className="mt-3 py-3"
    noGutters={false}
  >
    <Col
      className="col-md-12 text-center"
    >
      <h4
        className="color-text"
      >
        <strong>
          Welcome to Counters
        </strong>
      </h4>
      <p
        className="color-text"
      >
        Capture cups of lattes, frappucinos, or anything else that can be counted.
      </p>
    </Col>
  </Row>
  <Row
    className="mt-2 py-2"
    noGutters={false}
  >
    <Col
      className="col-md-12 text-center py-32"
    >
      <Button
        onClick={[Function]}
      >
        Get started
      </Button>
    </Col>
  </Row>
</Container>
`;
